core/adapters/eth_tx.go

```diff
@@ -188,6 +188,10 @@ func ensureTxRunResult(input models.RunResult, str *strpkg.Store) models.RunResu
 var zero = common.Hash{}
 
 func addReceiptToResult(receipt *models.TxReceipt, input models.RunResult, output *models.RunResult) {
+	if receipt == nil {
+		return
+	}
+
```

Comment:
My naive impression is calling addReceiptToResult(...) with nil should be considered an error.  Maybe add shouldn't be called at all?

Reply:
Unfortunately I believe this is not ideal.  If a function is getting passed unexpected parameters returning an error or failing are the safest things to do.  Silently ignoring the system being in faulty state could be bad.

Reply:
@felder-cl you're right that this isn't ideal, but in this case we added recently, we return the [safe status along with an error](https://github.com/smartcontractkit/chainlink/blob/develop/core/store/tx_manager.go#L431). So I think this is an expected case.

Reply:
I am probably missing context, but.. I believe function's should only consider their arguments and purpose.  It will help avoid misuse or misreading of the source code.  Given we're aiming for stability, maybe an alternate approach is better?

Or another approach is renaming the function to tryAddReceipt..()? It indicates it will fail over silently on what looks like an error.

Reply:
@se3000 is right about `receipt == nil` being an expected case, so we don't need to surface an error.  But I also agree with @felder-cl's point about semantics and style.  It's not obvious that the function can sometimes be a no-op.  On balance, I think it feels cleaner to simply shift the `if` conditional outside of the function in the two places it's called (rather than prefixing with `try`).